cl/opinion_page/utils.py

```diff
@@ -1,24 +1,42 @@
 import csv
+import logging
+import traceback
 from io import StringIO
 from typing import Dict, Tuple, Union
 
 from asgiref.sync import sync_to_async
+from django.conf import settings
 from django.contrib.auth.models import AnonymousUser, User
 from django.core.cache import caches
 from django.core.exceptions import ObjectDoesNotExist
 from django.http import HttpRequest
 from django.shortcuts import aget_object_or_404  # type: ignore[attr-defined]
-from elasticsearch_dsl import Q
+from django_elasticsearch_dsl.search import Search
+from elasticsearch.exceptions import ApiError, ConnectionTimeout, RequestError
+from elasticsearch_dsl import MultiSearch, Q
 
 from cl.alerts.models import DocketAlert
 from cl.custom_filters.templatetags.text_filters import best_case_name
 from cl.favorites.forms import NoteForm
 from cl.favorites.models import Note
-from cl.lib.elasticsearch_utils import do_count_query
+from cl.lib.bot_detector import is_bot
+from cl.lib.elasticsearch_utils import (
+    build_es_main_query,
```

Comment:
It seems like `build_es_main_query` might be unused in this file.

Reply:
removed